### PR TITLE
[iOS] Fixed a clipping bug with RenderTransform, mostly affecting <ViewBox>

### PR DESCRIPTION
**Pull request by carldebilly**
Fixes https://github.com/unoplatform/uno/issues/4918
Fixes https://github.com/unoplatform/uno/issues/4976

# Bugfix
On iOS, the wrong coordinates were used to provide the `availableSize` for the _arrange_ phase. The correct iOS value to use is `.Bounds`, but `.Frame` were used.

`<ViewBox>` were the most affected control because it was often triggering a clipping with wrong measurements on the content control.

Another related fix is to disable clipping on controls when transforms are used.

## PR Checklist

Please check if your PR fulfills the following requirements:

- ~~[ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)~~
- [X] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [X] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

**Review by mauadi — APPROVED** [review]

File: src/Uno.UI/Extensions/UIViewExtensions.iOSmacOS.cs [hunk 1]
```diff
@@ -651,9 +664,11 @@ StringBuilder AppendView(_View innerView)
 						.Append(fe != null && fe.TryGetPadding(out var p) && p != default ? $" Padding={p}" : "")
 						.Append(fe != null && fe.TryGetCornerRadius(out var cr) && cr != default ? $" CornerRadius={cr.ToStringCompact()}" : "")
 						.Append(uiElement?.Clip != null ? $" Clip={uiElement.Clip.Rect}" : "")
-						.Append(uiElement != null ? $" DesiredSize={uiElement.DesiredSize}, AvailableSize={uiElement.LastAvailableSize}" : "")
+						.Append(uiElement != null ? $" AvailableSize={uiElement.LastAvailableSize}" : "")
 						.Append(uiElement?.NeedsClipToSlot ?? false ? " CLIPPED_TO_SLOT" : "")
-						.Append(innerView is TextBlock textBlock ? $" Text=\"{textBlock.Text}\"" : "")
+						.Append(innerView is TextBlock textBlock ? $" Text=\"{textBlock.Text}\" - {textBlock.FontFamily}/{textBlock.FontSize}" : "")
+						.Append(innerView is Viewbox viewBox ? $" Scale={viewBox.scaleX}x{viewBox.scaleY} - Stretch={viewBox.Stretch}/{viewBox.StretchDirection}" : "")
```
Comment (dr1rrb) [inline src/Uno.UI/Extensions/UIViewExtensions.iOSmacOS.cs:670]: Redundant with the `tranforms` which might be really confusing.

File: src/Uno.UI/UI/Xaml/Controls/ViewBox/Viewbox.cs [hunk 2]
```diff
@@ -18,6 +18,8 @@ namespace Windows.UI.Xaml.Controls
 	public partial class Viewbox : global::Windows.UI.Xaml.FrameworkElement
 	{
 		private readonly Border _container;
+		internal double scaleX { get; private set; }
+		internal double scaleY { get; private set; }
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Controls/ViewBox/Viewbox.cs:22]: Invalid casing for a property :)

File: src/Uno.UI/UI/Xaml/Controls/ViewBox/Viewbox.cs [hunk 3]
```diff
@@ -55,13 +57,26 @@ protected override Size MeasureOverride(Size availableSize)
...
-			};
+				_container.RenderTransform = null;
+			}
+			else
+			{
+				var transform = _container.RenderTransform as ScaleTransform ?? new ScaleTransform();
+				transform.ScaleX = scaleX;
+				transform.ScaleY = scaleY;
+				transform.CenterX = finalSize.Width / 2d;
+				transform.CenterY = finalSize.Height / 2d;
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Controls/ViewBox/Viewbox.cs:77]: You could have set the `RenderTransformOrigin` is there a specific reason to do it like that?

File: src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.cs [hunk 4]
```diff
@@ -188,7 +188,13 @@ public void Arrange(Rect finalRect)
 				bool allowClipToSlot;
 				bool needsClipToSlot;
 
-				if (Panel is ICustomClippingElement customClippingElement)
+				if (Panel.RenderTransform != null)
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.cs:191]: If a view has *any* RenderTransform at all it doesn't get clipped?!

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**